### DISCUSSION TITLE
Prioritization of the APIs

### DISCUSSION THREAD

**Discussion by sandersms** [opiproject/opi-api#5] [Ideas]
Infrastructure needs to be the first step in the API definitions and we need to consider the differences between control plane and data/user plane traffic.

From a prioritization standpoint we seemed to align on the following set:

1. Networking API - Standardize on the OpenConfig model for configuration
2. Security API - Focus on the definition of what is needed for a Firewall Use Case to align with Use Case and PoC group plans
3. Storage API - this will require definition work
4. AI/ML - High level ideas around how to create this API, and what can be leveraged from any existing standards

Most of this set above primarily covers configuration.
How do we define the consumption interface that the Data/User plane will utilize?

Is this the proper priority set or is there something else we should focus on?

Do we need to build a gRPC concept to start creating the configuration infrastructure against?

**Comment by jerinjacobk** [comment]
> Infrastructure needs to be the first step in the API definitions and we need to consider the differences between control plane and data/user plane traffic.
> 
> From a prioritization standpoint we seemed to align on the following set:
> 
> 1. Networking API - Standardize on the OpenConfig model for configuration

There are a lot of protocols in open config, I guess, We need to decide the use case first, that, we need to tackle  and associated protocols and leverage config from openConfig.

> 2. Security API - Focus on the definition of what is needed for a Firewall Use Case to align with Use Case and PoC group plans

Marvell thinks the TLS use case could be one to start with.

> 3. Storage API - this will require definition work
> 4. AI/ML - High level ideas around how to create this API, and what can be leveraged from any existing standards
> 
> Most of this set above primarily covers configuration. How do we define the consumption interface that the Data/User plane will utilize?
> 
> Is this the proper priority set or is there something else we should focus on?
> 
> Do we need to build a gRPC concept to start creating the configuration infrastructure against?

Yes. I believe, we need to define .proto files for services for the control plane. For dataplane, We need some framework to use userplane messages.
